src/sentry/static/sentry/app/components/forms/formField.jsx

```diff
@@ -112,8 +112,9 @@ export default class FormField extends React.PureComponent {
         value,
       },
       () => {
-        this.props.onChange && this.props.onChange(this.coerceValue(this.state.value));
-        form && form.onFieldChange(this.props.name, this.coerceValue(this.state.value));
+        let finalValue = this.coerceValue(this.state.value);
```

Comment:
note: this is the reason all logic can't go into coerceValue: we only send results of coerceValue upstream, we dont store them with form state